hexlify.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Module to add and remove Python scripts into and from a MicroPython hex.
+ */
+var upyhex = (function() {
+    'use strict';
+
+    /** User script located at specific flash address. */
+    var USER_CODE_START_ADDR = 0x3e000;
+    var USER_CODE_LEN = 8 * 1024;
+    var USER_CODE_END_ADDR = USER_CODE_START_ADDR + USER_CODE_LEN;
+
+    /** User code header */
+    var USER_CODE_HEADER_SIZE = 4;
+    var USER_CODE_HEADER_START_B0_INDEX = 0;
+    var USER_CODE_HEADER_START_B1_INDEX = 1;
+    var USER_CODE_HEADER_LEN_LSB_INDEX = 2;
+    var USER_CODE_HEADER_LEN_MSB_INDEX = 3;
+
+    /** Number of data bytes per Intel Hex record (line). */
+    var INTEL_HEX_BYTE_CNT = 16;
+
+    /** Start of user script marked by "MP" + 2 bytes for the script length. */
+    var USER_CODE_HEADER_START_B0 = 77;    // 'M'
+    var USER_CODE_HEADER_START_B1 = 80;    // 'P'
+
+    /**
+     * String placed inside the MicroPython hex string to indicate where to
+     * paste the Python Code
+     * */
+    var HEX_INSERTION_POINT = ":::::::::::::::::::::::::::::::::::::::::::\n";
+
+    /**
+     * Converts a string into a byte array of characters.
+     * TODO: Update to encode to UTF-8 correctly.
```

Comment:
Issue #60

Reply:
Thanks Ross, I also created https://github.com/microbit-foundation/js-microbit-micropython-fs/issues/1 in the new filesystem library to capture this issue.
Marking this comment as resolved as fixing it is currently out of the scope for this PR.